server/controllers/profile.js

```diff
@@ -0,0 +1,44 @@
+const Profile = require("../models/Profile");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+
+// @route POST /profile/edit
+// @desc edit user profile
+// @access Public
+exports.editProfile = asyncHandler(async (req, res, next) => {
+  const user = await User.findById(req.user.id);
+
+  if (!user) {
+    res.status(404);
+    throw new Error("User doesn't exist");
+  }
+  try {
```

Comment:
We don't a try/catch as the `asyncHandler` will do the error handling for us and pass it to the error middleware